### PR TITLE
Remove Emergency address from managers

### PR DESCRIPTION
**Pull request by darosior**
This updates to latest revault_tx and revault_net to finally be able to prevent the manager to create Emergency transactions.
Note that in `test_sig_sharing` was *not* modified to only check stakehodlers pass to `SECURED` state: managers will implicitly pass to the SECURED state as all their presigned transactions are now signed (their Emergency transactions were never created in the first place).
The SECURED state does not make much sense for stakeholders but it will ease the passage to ACTIVE later on.

## Warning

This broke the `listtransactions` API as a result of not allowing manager to be able to create the transactions at all. `listtransactions` is fucked up anyways and should be burnt with fire, but still, sorry for the breakage Ed!

### CODE REVIEW

**Review by edouardparis — COMMENTED** [review]
Github does not make it clear what are the change you did in `coordinatord`

**Comment by darosior** [comment]
> Github does not make it clear what are the change you did in coordinatord

Yes, sorry about that: i god-mode merged the changes in https://github.com/re-vault/coordinatord/pull/5 , so this is just an update to coordinatord's master.

File: src/common/config.rs [hunk 1]
```diff
@@ -1,8 +1,56 @@
...
 
-use revault_tx::bitcoin::{util::bip32, Address, Network, PublicKey};
+use revault_net::noise::PublicKey as NoisePubKey;
+use revault_tx::bitcoin::{hashes::hex::FromHex, util::bip32, Address, Network, PublicKey};
 
-use serde::Deserialize;
+use serde::{de, Deserialize};
+
+#[derive(Debug, Clone)]
+pub struct NoisePubkeyHex {
```
[the remark this answers is not in the document]
Comment (darosior) [reply src/common/config.rs:9]: Yea, but it's just really a hack. I'm considering proposing a patch to sodiumoxide to implement a string visitor in their serde implem
Let's keep it as is for now

File: src/common/config.rs [hunk 2]
```diff
@@ -26,6 +74,7 @@ pub struct WatchtowerConfig {
 pub struct StakeholderConfig {
     pub xpub: bip32::ExtendedPubKey,
     pub watchtowers: Vec<WatchtowerConfig>,
+    pub emergency_address: Address,
```
Comment (edouardparis) [inline src/common/config.rs:77]: nit: Change address for revault_tx::EmergencyAddress using a serde Deserializer
example: https://github.com/re-vault/revault-gui/blob/master/src/revaultd/model.rs#L131

File: src/daemon/revaultd.rs [hunk 3]
```diff
@@ -344,7 +342,11 @@ impl RevaultD {
             config.unvault_csv,
         )?;
         let cpfp_descriptor = cpfp_descriptor(managers_pubkeys)?;
-        let emergency_address = EmergencyAddress::from(config.emergency_address)?;
+        let emergency_address = if let Some(conf) = config.stakeholder_config {
+            Some(EmergencyAddress::from(conf.emergency_address)?)
```
Comment (edouardparis) [inline src/daemon/revaultd.rs:346]: I think it is better to throw an error during the deserialization of the config than in the daemon. see https://github.com/re-vault/revaultd/pull/84/files#r576995072

Comment (darosior) [reply src/daemon/revaultd.rs:346]: I agree, i was being lazy :) Will amend.

File: src/common/config.rs (hunk 2 above)
Comment (darosior) [reply src/common/config.rs]: Added a commit upstream for this, as we may use it in other projects too.  See https://github.com/re-vault/revault_tx/pull/58#issuecomment-781419432

**Comment by darosior** [comment]
Updated upstream, we now parse the Emergency address from config! :tada:

**Comment by darosior** [comment]
Update to revault_tx master

**Review by edouardparis — APPROVED** [review]
ACK 7a69884eef00cdd1a9df70b3e0ca3574052c2e8d

**Pull request merged**